test/e2e/validation/sslprofile.go

```diff
@@ -63,17 +64,26 @@ func ValidateDefaultSslProfiles(ic *v1alpha1.Interconnect, f *framework.Framewor
 // its pods, querying management API for sslProfiles. Next it ensure that all sslProfile
 // definitions fro the sslProfMap are defined on each pod.
 func ValidateSslProfileModels(ic *v1alpha1.Interconnect, f *framework.Framework, sslProfMap SslProfileMapByName) {
+	var podNames []string
 
 	// Retrieve lastest version of given Interconnect resource
 	ic, err := f.GetInterconnect(ic.Name)
 	gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
 	// Validate IC instance
 	gomega.Expect(ic).NotTo(gomega.BeNil())
-	gomega.Expect(len(ic.Status.PodNames)).To(gomega.BeNumerically(">", 0))
 
-	// Iterating through all pods
-	for _, pod := range ic.Status.PodNames {
+	pods, err := f.KubeClient.CoreV1().Pods(ic.Namespace).List(metav1.ListOptions{LabelSelector: "application=" + ic.Name + ",interconnect_cr=" + ic.Name})
```

Comment:
Just out of curiousity, what was the reason for switching from using the interconnect status to listing pods directly?

Reply:
The test was failing intermittently when the ic.Status.PodNames was not populated. Thought it to be a timing issue with regard to the status update occurring. Switching to the filtered list did not exhibit any failures.